nodejs/eks/addon.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+import { Cluster, ClusterInternal } from "./cluster";
+
+/**
+ * AddonOptions describes the configuration options available for the Amazon VPC CNI plugin for Kubernetes. This is obtained from
+ * upstream `aws.eks.AddonArgs` and removes the deprecated `resolveConflicts` field. `clusterName` is also removed as we enable users to
+ * pass in the cluster object directly.
+ */
+export interface AddonOptions extends Omit<aws.eks.AddonArgs, "resolveConflicts" | "clusterName"> {
+    cluster: Cluster | ClusterInternal;
```

Comment:
Out of curiosity, would I be able to use an `aws.eks.Cluster` here?

I can see a use case where maybe you didn't start with pulumi-eks but there's still some convenience it can provide on top of what you currently have. Edit: although looking at it more, it's not doing much on top so this wouldn't be very valuable.